lib/buffer.js

```diff
@@ -272,30 +272,62 @@ Buffer.concat = function(list, length) {
 };
 
 
+function base64ByteLength(str, bytes) {
+  // Handle padding
+  if (str.charCodeAt(bytes - 1) === 0x3D)
+    bytes--;
+  if (bytes > 1 && str.charCodeAt(bytes - 1) === 0x3D)
+    bytes--;
+
+  // Base64 ratio: 3/4
+  return (bytes * 3) >>> 2;
+}
+
+
 function byteLength(string, encoding) {
-  if (typeof(string) !== 'string')
-    string = String(string);
+  if (typeof string !== 'string')
+    string = '' + string;
```

Comment:
@bnoordhuis do we care if node aborts if someone does:
```js
process.binding('buffer').byteLengthUtf8(42);
```
Or do we want to continue throwing?